src/azure-account.ts

```diff
@@ -376,12 +380,23 @@ export class AzureLoginHelper {
 		if (selected) {
 			const config = workspace.getConfiguration('azure');
 			if (config.get('cloud') !== selected.environment.name) {
-				this.doLogin = true;
+				if (selected.environment.name === azureCustomCloud) {
+					const armUrl = await window.showInputBox({
+						prompt: localize('azure-account.enterArmUrl', "Enter the Azure Resource Manager endpoint"),
+						placeHolder: 'https://management.local.azurestack.external',
+						ignoreFocusOut: true
+					});
+					if (armUrl) {
+						await config.update(customCloudArmUrlKey, armUrl, getCurrentTarget(config.inspect(customCloudArmUrlKey)));
+					} else {
+						return;
+					}
+				}
+
 				// if outside of normal range, set ppe setting
-				config.update('cloud', selected.environment.name, getCurrentTarget(config.inspect('cloud')));
-			} else {
-				return this.login('loginToCloud');
+				await config.update('cloud', selected.environment.name, getCurrentTarget(config.inspect('cloud')));
 			}
+			return this.login('loginToCloud');
```

Comment:
Hey @RMacfarlane , I did some investigation, I think this doLogin is used to let us call logIn function inside initial function when we aren't logged. If we set multiple this.doLogIn = true, we will call logIn function multiple times, and will also trigger serval login pages. And I don't think it is reasonable if we call logIn function when we ain't finish setting the user setting. So I just change this piece of code a little bit just call logIn function in the end after we finish our config updates. What do you think about it ?

Reply:
Thanks, this is a good solution